R/module_nested_tabs.R

```diff
@@ -368,3 +374,41 @@ srv_teal_module.teal_module <- function(id,
     )
   }
 }
+
+#' Calls expression when condition is met
+#'
+#' Function postpones `handlerExpr` to the moment when `eventExpr` (condition) returns `TRUE`,
+#' otherwise nothing happens.
+#' @param condExpr logical expression determining moment when `handlerExpr` should be evaluated.
+#' @param ... additional arguments passed to `observeEvent` with the exception of `eventExpr` that is not allowed.
+#' @inheritParams shiny::observeEvent
+#'
+#' @return An observer.
+#'
+#' @keywords internal
+call_once_when <- function(condExpr,
+                           handlerExpr,
+                           event.env = parent.frame(),
+                           handler.env = parent.frame(),
+                           ...) {
+  if ("eventExpr" %in% names(rlang::list2(...))) {
+    stop("eventExpr is not allowed in call_once_when.")
+  }
```

Comment:
Alternatively, we can change arg name `condExpr` -> `eventExpr`, but I don't mind so much this one.

Reply:
I wanted to keep a different name to avoid confusions as it needs to be a logical expression.

But given it's not an exported function, I think we can use `eventExpr`.

Done in 126c54710